subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/DefaultIgnoredConfigurationInputs.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configurationcache
+
+import org.gradle.configurationcache.initialization.ConfigurationCacheStartParameter
+import org.gradle.internal.service.scopes.Scopes.BuildTree
+import org.gradle.internal.service.scopes.ServiceScope
+import org.gradle.util.internal.GFileUtils
+import java.io.File
+import javax.inject.Inject
+
+
+@ServiceScope(BuildTree::class)
+class DefaultIgnoredConfigurationInputs(
+    ignoredPathsString: String?,
+    private val rootDirectory: File
+) : IgnoredConfigurationInputs {
+
+    @Inject
+    @Suppress("unused") // used in DI
+    constructor(configurationCacheStartParameter: ConfigurationCacheStartParameter) :
+        this(configurationCacheStartParameter.ignoredFileSystemCheckInputs, configurationCacheStartParameter.rootDirectory)
+
+    private
+    val userHome = File(System.getProperty("user.home"))
+
+    private
+    val jointRegex: Regex? = maybeCreateJointRegexForPatterns(ignoredPathsString)
+
+    override fun isFileSystemCheckIgnoredFor(file: File): Boolean =
+        jointRegex?.matches(normalizeActualInputPath(file)) ?: false
```

Comment:
We probably need to handle case sensitivity here :( `org.gradle.internal.nativeintegration.filesystem.FileSystem.isCaseSensitive` seems to be a good enough starting point, though it doesn't handle multiple mount points well and relies on the case sensitivity of `java.io.tempdir` location.

Reply:
Done with `FileSystem.isCaseSensitive` in https://github.com/gradle/gradle/compare/53fafc84165ddee58c516687f109d369953b6be5..95aa50e9a91b0c5bfe13f54c71d99ca340d517f9

I don't think there is a good way to handle multiple file systems, as a pattern may spread across file systems, so we can only try to do something based on the actual `File` (like, get the corresponding `java.nio.FileSystem` and make it produce a `PathMatcher` which we can abuse to check if the filesystem is case-sensitive). But this won't work reliably if the file's filesystem changes between the calls.

Reply:
Let's assume `FileSystem.isCaseSensitive` is enough for now.